plugin/virustotal/vtreport.py

```diff
@@ -0,0 +1,404 @@
+    ip_details.append(ip_analysis)
+    ip_details.append(ip_report.get('country'))
+    return ip_details
+
+  def get_contacted_urls(self):
+    return self.contacted_urls.keys()
+
+  def get_contacted_url(self, url):
+    """Return a list containing URL + detections."""
+
+    url_report = self.contacted_urls[url]
+    url_details = []
+    url_analysis = url_report.get('last_analysis_stats')
+    total = url_analysis['harmless'] + url_analysis['malicious']
+    total += url_analysis['suspicious'] + url_analysis['undetected']
+    url_analysis = str(url_analysis['malicious']) + '/' + str(total)
+    url_details.append(url)
+    url_details.append(url_analysis)
+    url_details.append(url_report.get('last_analysis_date'))
+    return url_details
+
+  def get_embedded_urls(self):
+    return self.embedded_urls.keys()
+
+  def get_embedded_url(self, url):
+    """Return a list containing URL + detections + scanned time."""
+
+    url_report = self.embedded_urls[url]
+    url_details = []
+    url_details.append(url)
+    detections = url_report.get('last_analysis_stats')
+
+    if detections:
+      total = detections['harmless'] + detections['malicious']
+      total += detections['suspicious'] + detections['undetected']
+      ratio = str(detections['malicious']) + '/' + str(total)
+    else:
+      ratio = None
+
+    url_details.append(ratio)
+    url_details.append(url_report.get('last_analysis_date'))
+    return url_details
+
+  def get_contacted_domains(self):
+    return self.contacted_domains.keys()
+
+  def get_contacted_domain(self, domain):
+    """Return a list containing IP + detections."""
+
+    domain_report = self.contacted_domains[domain]
+    domain_details = []
+    domain_analysis = domain_report.get('last_analysis_stats')
+    total = domain_analysis['harmless'] + domain_analysis['malicious']
+    total += domain_analysis['suspicious'] + domain_analysis['undetected']
+    domain_analysis = str(domain_analysis['malicious']) + '/' + str(total)
+    domain_details.append(domain)
+    domain_details.append(domain_analysis)
+    domain_details.append(domain_report.get('creation_date'))
+    domain_details.append(domain_report.get('registrar'))
+
+    return domain_details
+
+  def get_embedded_domains(self):
+    return self.embedded_domains.keys()
+
+  def get_embedded_domain(self, domain):
+    """Return a list containing IP + detections."""
+
+    domain_report = self.embedded_domains[domain]
+    domain_details = []
+    domain_analysis = domain_report.get('last_analysis_stats')
+    total = domain_analysis['harmless'] + domain_analysis['malicious']
+    total += domain_analysis['suspicious'] + domain_analysis['undetected']
+    domain_analysis = str(domain_analysis['malicious']) + '/' + str(total)
+    domain_details.append(domain)
+    domain_details.append(domain_analysis)
+    domain_details.append(None)
+    domain_details.append(domain_report.get('creation_date'))
+    domain_details.append(None)
```

Comment:
A extra final comment: it feels to me as if these functions are creating a specific return structure because of the specific needs of other part of the code. If thats the case, maybe it makes more sense to split into standarized Model - View - Controller paradigm:

1. Keep in this class only as a Model, with accessor functions to return the full dict of a domain.
2. Add a separate "View" class or "View" function.